### PR TITLE
Add irreducible mass

### PR DESCRIPTION
**Pull request by jennifersanchez**
add Irreducible Mass

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by jennifersanchez** [comment]
I think we've fixed everything that you've asked us to do.

**Comment by geoffrey4444** [comment]
We’ll figure out the conflicts then squash

**Comment by nilsvu** [comment]
@jennifersanchez could you reopen and push to https://github.com/sxs-collaboration/spectre/pull/1260 instead please?

**Comment by geoffrey4444** [comment]
3 more quick things, because some tests are failing:

1. Line 33 of docs/References.bib needs a comma at the end; otherwise, the doc-check test fails because bibtex can't compile the bibliography for the help pages.

2. Please add again (it must have been removed accidentally), right after line 656 of src/ApparentHorizons/StrahlkorperGr.cpp, the following line: `ASSERT(area > 0.0, "The area of the horizon must be greater than zero but is " << area);`

3. Your new and improved test, comparing the Kerr result to your code's result, fails, because the results disagree at roundoff levels. Fix by changing line 285 of `tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp` from

`CHECK(irreducible_mass == expected_irreducible_mass);`

to

`CHECK(irreducible_mass == approx(expected_irreducible_mass));`

and see if that fixes it. You can compile locally and check that the tests all pass in docker.

I'm happy for you to squash after you make these 3 fixes.

UPDATE: I originally suggested this line, which @fmahebert corrected in the comment below (don't do this):

`CHECK_ITERABLE_APPROX(irreducible_mass, expected_irreducible_mass);`

**Comment by fmahebert** [comment]
Regarding @geoffrey4444's item 3 above:

`irreducible_mass` is a `double`, not a `DataVector`, so the new code should read:
```
CHECK(irreducible_mass == approx(expected_irreducible_mass));
```

**Comment by geoffrey4444** [comment]
Thanks, @fmahebert! Would my code not work at all, or is it just not the best way to do it?

**Comment by fmahebert** [comment]
It would work (I just tested it). But it implies the quantity is an iterable, so I find it clearer to use the simple CHECK.

**Review by kidder — COMMENTED** [review]
This looks good, but there is a conflict to be fixed in `References.bib`.

File: src/ApparentHorizons/StrahlkorperGr.cpp
Comment (fmahebert) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: this line needs clang-format

File: docs/References.bib
Comment (nilsleiffischer) [inline docs/References.bib]: Please address this comment from #1260:

> @jennifersanchez Since we last reviewed this PR, we have added guidelines on how to cite publications in the documentation. Please have a look at the section in https://spectre-code.org/writing_good_dox.html and format your bib entry and documentation accordingly.

In particular:
- Capitalize `Hartle2003gravity`
- 2 spaces indentation for the keys, then align all `=` like in `HesthavenWarburton`
- Use `{}` instead of `""` for `url`

Comment (nilsvu) [inline docs/References.bib]: Please address this comment from #1260:

> @jennifersanchez Since we last reviewed this PR, we have added guidelines on how to cite publications in the documentation. Please have a look at the section in https://spectre-code.org/writing_good_dox.html and format your bib entry and documentation accordingly.

In particular:
- Capitalize `Hartle2003gravity`
- 2 spaces indentation for the keys, then align all `=` like in `HesthavenWarburton`

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Review by geoffrey4444 — DISMISSED** [review]

**Comment by geoffrey4444** [comment]
Travis failures are timeouts unrelated to this PR. @nilsleiffischer do you approve? Ping @sxs-collaboration/spectre-core-devs

File: src/ApparentHorizons/StrahlkorperGr.hpp
Comment (nilsvu) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: Please match the key in `References.bib`: `Hartle2003gravity`

**Comment by nilsdeppe** [comment]
Restart both stage 1 and stage 2 clang-release and gcc6-debug builds because of timeouts.

@geoffrey4444 generally PRs should not time out. I'm not sure how you were able to evaluate that the timeout is unrelated.

I'm fine with this once the tests pass.

**Review by nilsvu — CHANGES_REQUESTED** [review]
You can squash this directly. (Generally you should push the changes you make in response to reviews as separate commits that begin with the word "fixup" to make them easier to review. Then squash them once the reviews have approved.)

**Comment by geoffrey4444** [comment]
@nils I looked at the Travis logs and they didn’t appear to get stuck on anything the PR touched, so I concluded the timeouts are probably not our fault. Looks like the rerun tests passed, so maybe that means I was right?

**Comment by geoffrey4444** [comment]
Thanks for the catches and comments, @nilsdeppe and @nilsleiffischer! We’ll take care of this last fix and squash

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsleiffischer please look at

**Review by geoffrey4444 — APPROVED** [review]

**Comment by geoffrey4444** [comment]
@nilsleiffischer can you please approve or let us know if you'd like further changes?

**Review by nilsvu — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by geoffrey4444** [comment]
@nilsdeppe is this ready to merge? Edit: you beat me to it :)

**Pull request merged**
